bus-mapping/src/evm/opcodes.rs

```diff
@@ -512,15 +516,27 @@ fn dummy_gen_call_ops(
     geth_steps: &[GethExecStep],
 ) -> Result<Vec<ExecStep>, Error> {
     let geth_step = &geth_steps[0];
-    let exec_step = state.new_step(geth_step)?;
+    let mut exec_step = state.new_step(geth_step)?;
 
-    let call = state.call()?.clone();
-    let callee = state.parse_call(geth_step)?;
+    let tx_id = state.tx_ctx.id();
+    let call = state.parse_call(geth_step)?;
 
-    let (_, account) = state.sdb.get_account(&callee.address);
+    let (_, account) = state.sdb.get_account(&call.address);
     let callee_code_hash = account.code_hash;
 
-    state.push_call(callee.clone(), geth_step);
+    let is_warm_prev = !state.sdb.add_account_to_access_list(call.address);
+    state.push_op_reversible(
```

Comment:
`push_op_reversible` internally calls `apply_op` for write ops, which itself applies the statedb change:
https://github.com/appliedzkp/zkevm-circuits/blob/344d2ea409c875f9c3fc792f9910423f172f35f9/bus-mapping/src/circuit_input_builder.rs#L1174

So I think with this implementation the statedb is written twice.
I think the line to get the `is_warm_prev` should be a read-only query to the statedb (`check_account_to_access_list`, which currently doesnt' exist), and we can leave the write in `push_op_reversible`.

Maybe we could rename `push_op_reversible` to `push_op_reversible_and_apply` to make it more explicity?

Reply:
I didn't know `push_op_reversible` calls `apply_op` :(  I have reverted the changes and use read-only methods. As for naming, I think it's fine to keep it as is.